matrix_sdk_base/src/client.rs

```diff
@@ -673,8 +756,8 @@ impl BaseClient {
     pub async fn receive_joined_timeline_event(
         &self,
         room_id: &RoomId,
-        event: &mut EventJson<RoomEvent>,
-    ) -> Result<(Option<EventJson<RoomEvent>>, bool)> {
+        event: &mut EventJson<AnyRoomEventStub>,
```

Comment:
The crypto stuff needs the room_id, or at least from my skimming and not fully understanding, it seemed to need a room_id. The `OlmMachine::decrypt_room_event` then returns an event with a room_id so I convert it back to a stub so the `BaseClient`, `Room` and `EventEmitter` methods are happy.

That does seem to add a lot of complexity just to avoid `Option<RoomId>`, at least for the clients. I'm sure there are improvements to what I've got here so maybe there is a more straightforward way of handing the events around that I'm not seeing.

If he wants to stick with the `Any*EventStub` for no `room_id` and `Any*Event` for thoes with we could just convert every event received in the sync response to a having a `room_id` then no switching back and forth from stub to normal.

Reply:
Well the decryption method could also take the room id as the argument. 

Though we do want to decrypt events that we get from the `/room/$roomId/messages` endpoint and since those are separate types we'll need two methods. Having two separate decryption methods doesn't sound nice either.